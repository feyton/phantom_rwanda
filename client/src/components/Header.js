import React, { forwardRef, useState } from 'react';
import { useTranslation } from 'react-i18next';
import { useSelector } from 'react-redux';
import { Link } from 'react-router-dom';
import phantom from '../assets/phantom.png';
import xmark from '../assets/xmark.svg';
import Burger from './Burger.js';
import Button from './Button.js';
import LanguageButton from './LanguageButton.js';
import MobileMenu from './MobileMenu.js';
import withClickOutside from './WithClickOutside.js';

const Header = forwardRef(({ open, setOpen }, ref) => {
  const { authenticated, user } = useSelector((state) => state?.auth);
  const [style, setStyle] = useState('hidden');
  const [icon, setIcon] = useState(<Burger />);
  const handleToggleMenu = () => {
    if (style === 'hidden') {
      setStyle('flex z-50');
      setIcon(<img src={xmark} alt="x mark" className="w-6" />);
    } else {
      setStyle('hidden');
      setIcon(<Burger />);
    }
  };
  const { t } = useTranslation();
  return (
    <div className="flex bg-background border-b-4 pb-4 sticky top-0 z-20">
      <div>
        <Link to="/">
          <img
            src="https://res.cloudinary.com/keza/image/upload/v1672036898/falcon-logo_1_wnbmct.png"
            alt="Phantom"
            className="w-44 xl:w-48 ml-8 xl:ml-20 mt-4"
          />
        </Link>
      </div>

      <div className="hidden md:flex xl:flex ml-auto mr-20">
        <Link to={!authenticated ? '/login' : '/dashboard'}>
          <Button
            name={!authenticated ? 'For operators' : 'Dashboard'}
            styles="bg-primary hover:bg-hover text-white rounded-xl mt-5 ml-6"
          />
        </Link>
        <Link to="/tracking-page">
          <Button
            name={t('track_bus')}
            styles="border border-primary bg-background hover:bg-hover2 text-primary rounded-xl mt-5 ml-6 px-6"
          />
        </Link>
        <LanguageButton />
      </div>
      <section
        ref={ref}
        className="flex flex-col absolute right-0 bg-background px-2"
      >
        <button
          type="button"
          id="toggle"
          className="md:hidden xl:hidden ml-auto mr-12 mt-5"
          onClick={() => setOpen(!open)}
        >
          {!open && <Burger />}
          {open && <img src={xmark} alt="xmark" className="w-6 mb-2" />}
        </button>

        {open && <MobileMenu />}
      </section>
    </div>
  );
});

export default withClickOutside(Header);
