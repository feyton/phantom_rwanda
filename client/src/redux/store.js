import { configureStore } from '@reduxjs/toolkit';
import { authReducer } from './reducers/authReducer.js';
import { busMovementReducer } from './reducers/busMovementReducer.js';
import { counterReducer } from './reducers/counterReducer.js';
import { dashboardReducer } from './reducers/dashboardReducer.js';
import { managementReducer } from './reducers/managementReducer.js';
import { profileReducer } from './reducers/profileReducer.js';
import { simulateReducer } from './reducers/simulateReducer.js';
import { socketReducer } from './reducers/socketReducer.js';
import { updateProfileReducer } from './reducers/updateProfileReducer.js';

export const store = configureStore({
  reducer: {
    counter: counterReducer,
    profileUpdates: updateProfileReducer,
    profileInfo: profileReducer,
    auth: authReducer,
    management: managementReducer,
    dashboard: dashboardReducer,
    socket: socketReducer,
    simulate: simulateReducer,
    movement: busMovementReducer
  }
});
