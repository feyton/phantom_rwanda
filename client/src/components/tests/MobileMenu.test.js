import React from 'react';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import { store } from '../../redux/store.js';
import Header from '../Header.js';

describe('MobileMenu', () => {
  it('should render Mobile menu', () => {
    const mobileMenu = renderer
      .create(
        <Provider store={store}>
          <BrowserRouter>
            <Header />
          </BrowserRouter>
        </Provider>
      )
      .toJSON();
    expect(mobileMenu).toMatchSnapshot();
  });
});
