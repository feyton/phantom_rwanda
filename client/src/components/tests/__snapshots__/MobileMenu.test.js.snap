// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MobileMenu should render Mobile menu 1`] = `
<div
  className="flex bg-background border-b-4 pb-4 sticky top-0 z-20"
>
  <div>
    <a
      href="/"
      onClick={[Function]}
    >
      <img
        alt="Phantom"
        className="w-44 xl:w-48 ml-8 xl:ml-20 mt-4"
        src="phantom.png"
      />
    </a>
  </div>
  <div
    className="hidden md:flex xl:flex ml-auto mr-20"
  >
    <a
      href="/login"
      onClick={[Function]}
    >
      <div>
        <button
          className="bg-primary hover:bg-hover text-white rounded-xl mt-5 ml-6 rounded-xl py-2 text-base md:text-sm px-5"
          type="submit"
        >
          For operators
        </button>
      </div>
    </a>
    <a
      href="/tracking-page"
      onClick={[Function]}
    >
      <div>
        <button
          className="border border-primary bg-background hover:bg-hover2 text-primary rounded-xl mt-5 ml-6 px-6 rounded-xl py-2 text-base md:text-sm px-5"
          type="submit"
        >
          track_bus
        </button>
      </div>
    </a>
    <div>
      <select
        className="bg-background border outline-none md:ml-6 lg:ml-6 xl:ml-6 mt-5 flex py-2 px-4 text-sm hover:font-bold"
        onChange={[Function]}
      >
        <option
          className=" border-none bg-background w-full cursor-pointer"
          value="en"
        >
          English
        </option>
        <option
          className=" border-none bg-background w-full cursor-pointer"
          value="fr"
        >
          French
        </option>
        <option
          className=" border-none bg-background cursor-pointer"
          value="kn"
        >
          Kinyarwanda
        </option>
      </select>
    </div>
  </div>
  <section
    className="flex flex-col absolute right-0 bg-background px-2"
  >
    <button
      className="md:hidden xl:hidden ml-auto mr-12 mt-5"
      id="toggle"
      onClick={[Function]}
      type="button"
    >
      <div>
        <div
          className="undefined h-1 w-7 bg-primary mt-1"
        />
        <div
          className="undefined h-1 w-7 bg-primary mt-1"
        />
        <div
          className="undefined h-1 w-7 bg-primary mt-1"
        />
      </div>
    </button>
  </section>
</div>
`;
